aoai_vs_oai_diffs.md

```diff
@@ -0,0 +1,81 @@
+## Diff between OAI and AOAI
+
+The differences detailed in this document were derived from comparing `{project_root}/.typespec/openapi.yaml` (OAI swagger file) and `Azure/azure-rest-api-spec/specifications/cognitiveservices/data-plane/AzureOpenAI/preview/2024-05-01-preview/inference.yaml` (AOAI latest release version for the SDKs)
+
+## Operations
+
+Listing differences in the request/response models between Azure and non-Azure
+
+### `/completions`
+
+- [x] `CreateCompletionRequest`
+  - [x] Additional Azure field: `completion_config`
+  - [x] non-Azure has `model`
```

Comment:
We can consider AOAI to still have `model` but make it optional and functionally ignored. That is: adding `"foo":" "bar"` to a request payload results in a parameter validation error, but adding `"model": "bar"` works just fine.

In general, I suspect that's the only semi-reasonable mechanism that a downstream API can use to "remove" an upstream property.

Reply:
Gotcha! Sorry that it seems like I am deriving an actionable item in this checklist. This is simply a list of the differences I found in the surface components of the swagger. I ticked `model` off because I don't plan on doing anything about it, just that it was "observed" as difference and it has been addressed if necessary.